File: examples/typescript/events/get-event-destination-not-found.ts
/**
 * Handle a destination ID that doesn't exist.
 *
 *   STEDI_API_KEY=<api-key> tsx get-event-destination-not-found.ts
 *
 * Asks for an ID that was never created, and fails if the 404 never arrives.
 */

import { randomUUID } from "node:crypto";

import { NotFoundException, Stedi } from "@stedi/sdk";

const MISSING_DESTINATION_ID = `dst_${randomUUID()}`;

/** Fetches the missing destination and returns the 404; anything else is a real failure. */
async function getExpectingNotFound(client: Stedi): Promise<NotFoundException> {
  try {
    await client.getEventDestination({ destinationId: MISSING_DESTINATION_ID });
  } catch (err) {
    if (err instanceof NotFoundException) {
      return err;
    }
    throw err;
  }
  throw new Error(`expected ${MISSING_DESTINATION_ID} not to exist`);
}

const apiKey = process.env["STEDI_API_KEY"];
if (!apiKey) {
  throw new Error("STEDI_API_KEY is not set");
}

const client = new Stedi({ apiKey: { apiKey } });
const notFound = await getExpectingNotFound(client);

console.log(`${MISSING_DESTINATION_ID} is not a destination in this account:`);
console.log(`  ${notFound.name}: ${notFound.message}`);

File: examples/python/events/get_event_destination_not_found.py
# /// script
# requires-python = ">=3.12"
# dependencies = ["stedi==0.0.10"]
# ///

"""Handle a destination ID that doesn't exist.

    STEDI_API_KEY=<api-key> uv run --script get_event_destination_not_found.py

Asks for an ID that was never created, and fails if the 404 never arrives.
"""

import asyncio
import os
from uuid import uuid4

from stedi import Config, Stedi
from stedi.models import GetEventDestinationInput, NotFoundException

MISSING_DESTINATION_ID = f"dst_{uuid4()}"


async def main(api_key: str) -> None:
    async with Stedi(Config(api_key=api_key)) as client:
        try:
            await client.get_event_destination(
                GetEventDestinationInput(destination_id=MISSING_DESTINATION_ID),
            )
        except NotFoundException as not_found:
            print(f"{MISSING_DESTINATION_ID} is not a destination in this account:")
            print(f"  {type(not_found).__name__}: {not_found.message}")
        else:
            raise SystemExit(f"expected {MISSING_DESTINATION_ID} not to exist")


api_key = os.environ.get("STEDI_API_KEY")
if not api_key:
    raise SystemExit("STEDI_API_KEY is not set")

asyncio.run(main(api_key))
